modules/exploits/multi/persistence/sshkey.rb

```diff
@@ -0,0 +1,313 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'sshkey'
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = ExcellentRanking
+
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Post::File
+  include Msf::Post::Unix
+  include Msf::Post::Windows::UserProfiles
+  include Msf::Exploit::EXE
+  include Msf::Exploit::Local::Persistence
+  include Msf::Exploit::Deprecated
+  moved_from 'post/linux/manage/sshkey_persistence'
+  moved_from 'post/windows/manage/sshkey_persistence'
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'SSH Key Persistence',
+        'Description' => %q{
+          This module will add an SSH key to a specified user (or all), to allow
+          remote login via SSH at any time. No payload is required for this module to work.
+
+          If an SSH key is not provided, a new 4096 bit RSA keypair will be generated.
+          The private key will be stored as loot for later use.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'h00die <mike@shorebreaksecurity.com>', # linux
+          'Dean Welch <dean_welch[at]rapid7.com>' # windows
+        ],
+        'Platform' => %w[linux win],
```

Comment:
```suggestion
        'Platform' => %w[unix win],
```

As this works on AIX/BSD/OSX/Solaris/… as well.

Reply:
I thought about this originally, and testing it on OSX, however since OSX has diff home directory locations so some of the assumptions in the module may not work. Also, my back log of persistence modules is long, so I wanted to KISS and just combine the two and let someone else expand if if they want to.